### DISCUSSION TITLE
Flutter / FCM notifications not arriving on Codemagic iOS build

### DISCUSSION THREAD

**Discussion by Aditya-QSS** [codemagic-ci-cd/codemagic-docs#2064] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I'm trying to build a Flutter app for both Android and iOS. I've been implementing push notifications using Firestore Cloud Messaging (FCM) and it works for both Android and iOS when I'm testing on a simulator (Android) or on a connected physical device (iOS).

However, when I built the app for the client to test (using Codemagic), the push notifications were not working on iOS (it does on Android). If it works on my development environment but not on Codemagic, everything is configured correctly except for Codemagic's build, right??

I did check the profile and the certificate generated by Codemagic and they were properly configured with Push Notifications. I set Codemagic after setting FCM and testing it on a local environment (and succeeding), so it should have worked I guess

### Expected results

Push Notification should come for both Android and iOS build.

### Actual results

Push Notification is coming in the Android build but not on the iOS build. However, I am getting push notifications when I am running the App on my real device but through codemagic build I am not getting push notifications.

**Comment by himesh-cm** [comment]
Hi @Aditya-QSS, do push notifications work when you build ipa locally and upload it to Testflight?

**Reply by himesh-cm** [reply]
Hi, @Aditya-QSS, Can you confirm if you are still facing this issue at your end?
If you want me to investigate the issue further, please share a build id.

**Reply by Aditya-QSS** [reply]
Hello @himesh-cm  Yes I am able to resolve the issue.

**Reply by himesh-cm** [reply]
That's great! Can you share with us what worked for you? It may help others facing similar issues.

**Reply by Aditya-QSS** [reply, marked as answer]
@himesh-cm  The issue is with the dev and prod environment firebase config mismatch.
